Model/Ui/AdyenOneclickConfigProvider.php

```diff
@@ -215,12 +215,15 @@ protected function getCcAvailableTypes()
     {
         $types = [];
         $ccTypes = $this->_adyenHelper->getAdyenCcTypes();
-        $availableTypes = $this->_adyenHelper->getAdyenCcConfigData('cctypes');
-        if ($availableTypes) {
-            $availableTypes = explode(',', $availableTypes);
-            foreach (array_keys($ccTypes) as $code) {
-                if (in_array($code, $availableTypes)) {
-                    $types[$code] = $ccTypes[$code]['name'];
+        $enableAvailableTypes = $this->_adyenHelper->getAdyenCcConfigData('enablecctypes');
+        if ($enableAvailableTypes) {
+            $availableTypes = $this->_adyenHelper->getAdyenCcConfigData('cctypes');
+            if ($availableTypes) {
+                $availableTypes = explode(',', $availableTypes);
+                foreach (array_keys($ccTypes) as $code) {
+                    if (in_array($code, $availableTypes)) {
+                        $types[$code] = $ccTypes[$code]['name'];
+                    }
                 }
             }
         }
```

Comment:
This is the exact same as in Model/Ui/AdyenCcConfigProvider.php. Would it make sense to outsource this code and reuse the function?

Reply:
@acampos1916 Is this the only commnet that is blocking us from merging?